src/d3d12/mod.rs

```diff
@@ -3,6 +3,43 @@ use log::{log, Level};
 use winapi::shared::winerror;
 use winapi::um::d3d12;
 
+#[cfg(feature = "public-winapi")]
+pub trait AbstractWinapiPtr<T> {
+    #[allow(clippy::mut_from_ref)]
+    fn as_winapi(&self) -> &mut T;
+}
+
+#[cfg(not(feature = "public-winapi"))]
+trait AbstractWinapiPtr<T> {
+    #[allow(clippy::mut_from_ref)]
+    fn as_winapi(&self) -> &mut T;
+}
```

Comment:
I am not happy with this solution, but I am not sure if we have a better alternative.
The duplicate code is not nice, and disabling clippy is almost always a bad idea.

Reply:
Perhaps something like this would work?
```suggestion
mod need_a_better_name {
    pub trait AbstractWinapiPtr<T> {
        #[allow(clippy::mut_from_ref)]
        fn as_winapi(&self) -> &mut T;
    }
}
#[cfg(feature = "public-winapi")]
pub use need_a_better_name::*;

#[cfg(not(feature = "public-winapi"))]
use need_a_better_name::*;
```

Reply:
Yep, great hack, blame Rust for this.